src/c-writer.cc

```diff
@@ -875,62 +946,162 @@ void CWriter::WriteFuncTypes() {
   Write(CloseBrace(), Newline());
 }
 
-void CWriter::WriteImports() {
-  if (module_->imports.empty())
+void CWriter::ComputeUniqueImports() {
+  std::map<std::pair<std::string, std::string>, const Import*> import_map;
+  for (const Import* import : module_->imports) {
+    import_map.try_emplace({import->module_name, import->field_name}, import);
```

Comment:
It's fine for a module to import the same external field under multiple ids (if the local fields are all the same type), but not so fine otherwise. Added a warning here on duplicate imports and an abort on clearly contradictory imports (https://github.com/fixpointOS/wabt/blob/module_instancing/src/c-writer.cc#L953-L962). I don't think this will bomb out on every contradictory import, though (e.g. if the same field is imported as two functions with different signatures).